src/rt-app_utils.c

```diff
@@ -31,6 +32,16 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 int log_level = 50;   // default
 
+#define C(x, v) { #x, LOG_LEVEL_ ## x},
+struct loglevel {
+	char *s;
+	int val;
+};
+
+struct loglevel lls[] = { LLS };
+#undef C
```

Comment:
I still don't like this undef then re-def. This is quite confusing and over complexify the code for no good reason. Furthermore, the "levels" are defined at 2 different places so you can't easily know what is it.
Gather everything related to level in one place and don't use undef

Reply:
i don't understand what you mean by that the levels are defined in 2 different places? clearly this is exactly the advantage of the above approach that you don't have to maintain 2 separate definitions, no? how would you do that without re-using one macro, i wonder...